tests/test_core.py

```diff
@@ -92,3 +92,57 @@ def test_measure_props(funcname, shape, chunks, has_lbls, ind):
     d_r = da_func(d, d_lbls, ind)
 
     dask_ndmeasure._test_utils._assert_eq_nan(a_r, d_r)
+
+
+@pytest.mark.parametrize(
+    "shape, chunks, ind", [
+        ((15, 16), (4, 5), None),
+        ((15, 16), (4, 5), 0),
+        ((15, 16), (4, 5), 1),
+        ((15, 16), (4, 5), [1]),
+        ((15, 16), (4, 5), [1, 2]),
+        ((15, 16), (4, 5), [1, 100]),
+    ]
+)
+@pytest.mark.parametrize(
+    "default", [
+        None,
+        0,
+        1.5,
+    ]
+)
+@pytest.mark.parametrize(
+    "pass_positions", [
+        False,
+        True,
+    ]
+)
+def test_labeled_comprehension(shape, chunks, ind, default, pass_positions):
+    a = np.random.random(shape)
+    d = da.from_array(a, chunks=chunks)
+
+    lbls = np.zeros(a.shape, dtype=np.int64)
+    lbls += (
+        (a < 0.5).astype(lbls.dtype) +
+        (a < 0.25).astype(lbls.dtype) +
+        (a < 0.125).astype(lbls.dtype) +
+        (a < 0.0625).astype(lbls.dtype)
+    )
+    d_lbls = da.from_array(lbls, chunks=d.chunks)
+
+    def func(val, pos=None):
+        if pos is None:
+            pos = 0 * val + 1
+
+        return (val * pos).sum() / (1 + val.max() * pos.max())
```

Comment:
The calls here to `max` would fail if the array is empty. This is exercised by the non-existent label case above (i.e. `100`). So this will fail if there is no data to compute on.